services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -0,0 +1,616 @@
+         */
+        Builder addNumberSet(String attributeName, Set<Number> values);
+
+        /**
+         * Appends an attribute of name attributeName with specified Set of byte array values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param values        Set of byte array values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addBinarySet(String attributeName, Set<byte[]> values);
+
+        /**
+         * Appends an attribute of name attributeName with specified  {@link Set}  of {@link ByteBuffer} values to the document
+         * builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param values        Set of ByteBuffer values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addByteBufferSet(String attributeName, Set<ByteBuffer> values);
+
+        /**
+         * Appends an attribute of name attributeName with specified binary byte array values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param values        The binary byte array values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addBinarySet(String attributeName, byte[]... values);
+
+        /**
+         * Appends an attribute of name attributeName with specified set of ByteBuffer values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param values        The ByteBuffer values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addBinarySet(String attributeName, ByteBuffer... values);
+
+        /**
+         * Appends an attribute of name attributeName with specified list of values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param value         The list of values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addList(String attributeName, List<?> value);
+
+        /**
+         * Appends an attribute of name attributeName with specified list of values to the document builder.
+         *
+         * @param attributeName Name of the attribute that needs to be added in the Document.
+         * @param values        The list of values that needs to be set.
+         * @return Builder instance to construct a {@link EnhancedDocument}
+         */
+        Builder addList(String attributeName, Object... values);
```

Comment:
Accepting a list of non type-safe values seems dangerous to me.

Reply:
Removed the var Args for all the API except AttributeConverters